gitlab/resource_gitlab_group_badge_test.go

```diff
@@ -0,0 +1,167 @@
+					}),
+				),
+			},
+			// Test ImportState
+			{
+				ResourceName:      "gitlab_group_badge.foo",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			// Update the group badge
+			{
+				Config: testAccGitlabGroupBadgeUpdateConfig(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabGroupBadgeExists("gitlab_group_badge.foo", &badge),
+					testAccCheckGitlabGroupBadgeAttributes(&badge, &testAccGitlabGroupBadgeExpectedAttributes{
+						LinkURL:  fmt.Sprintf("https://example.com/badge-%d", rInt),
+						ImageURL: fmt.Sprintf("https://example.com/badge-%d.svg", rInt),
+					}),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckGitlabGroupBadgeExists(n string, badge *gitlab.GroupBadge) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not Found: %s", n)
+		}
+
+		splitID := strings.Split(rs.Primary.ID, ":")
+
+		badgeID, err := strconv.Atoi(splitID[len(splitID)-1])
+		if err != nil {
+			return err
+		}
+		groupID := rs.Primary.Attributes["group"]
+		if groupID == "" {
+			return fmt.Errorf("No group ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*gitlab.Client)
+
+		gotBadge, _, err := conn.GroupBadges.GetGroupBadge(groupID, badgeID)
+		if err != nil {
+			return err
+		}
+		*badge = *gotBadge
+		return nil
+	}
+}
+
+type testAccGitlabGroupBadgeExpectedAttributes struct {
+	LinkURL  string
+	ImageURL string
+}
+
+func testAccCheckGitlabGroupBadgeAttributes(badge *gitlab.GroupBadge, want *testAccGitlabGroupBadgeExpectedAttributes) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if badge.LinkURL != want.LinkURL {
+			return fmt.Errorf("got link_url %q; want %q", badge.LinkURL, want.LinkURL)
+		}
+
+		if badge.ImageURL != want.ImageURL {
+			return fmt.Errorf("got image_url %s; want %s", badge.ImageURL, want.ImageURL)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckGitlabGroupBadgeDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*gitlab.Client)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "gitlab_group" {
+			continue
+		}
+
+		group, resp, err := conn.Groups.GetGroup(rs.Primary.ID, nil)
+		if err == nil {
+			if group != nil && fmt.Sprintf("%d", group.ID) == rs.Primary.ID {
+				if group.MarkedForDeletionOn == nil {
+					return fmt.Errorf("Group still exists")
+				}
+			}
+		}
+		if resp.StatusCode != 404 {
+			return err
+		}
+		return nil
+	}
+	return nil
+}
+
+func testAccGitlabGroupBadgeConfig(rInt int) string {
+	return fmt.Sprintf(`
+resource "gitlab_group" "foo" {
+  name        = "foo-%d"
+  path        = "foo-%d"
+  description = "Terraform acceptance tests"
+
+  # So that acceptance tests can be run in a gitlab organization
+  # with no billing
+  visibility_level = "public"
+}
+
+resource "gitlab_group_badge" "foo" {
+  group     = "${gitlab_group.foo.id}"
+  link_url  = "http://example.com/badge-%d"
```

Comment:
I am not sure if some point http:// connections will not be disabled to prevent mixed content I propose that you set `https://` here and make `testAccGitlabGroupBadgeUpdateConfig` function change the site url instead of protocol